common/src/main/java/net/modgarden/barricade/mixin/client/BarrierBlockMixin.java

```diff
@@ -10,6 +12,12 @@
 public class BarrierBlockMixin {
     @ModifyReturnValue(method = "getRenderShape", at = @At("RETURN"))
     private RenderShape barricade$barrierBlockModel(RenderShape original) {
+        var player = Minecraft.getInstance().player;
+
+        if (player == null || !player.canUseGameMasterBlocks() || !player.getMainHandItem().is(Items.BARRIER)) {
```

Comment:
```java
!player.getMainHandItem().is(Items.BARRIER)
```
This doesn't work for custom barriers that Barricade adds. Consider adding an item tag.